### PR TITLE
attempt to use system ROOT

### LINKED ISSUE #899: ROOT: build error on GHA
Currently [builds fail](https://github.com/SyneRBI/SIRF-SuperBuild/actions/runs/9070313125/job/24921825770#step:10:13597) with `BUILD_ROOT=ON` . This seems to be a transient error due to gremlins.

See:
- https://github.com/SyneRBI/SIRF-SuperBuild/pull/897#issuecomment-2109543050
- https://github.com/root-project/root/issues/14520

```
CMake Error at /home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/builds/ROOT/build/builtins/xrootd/XROOTD-prefix/src/XROOTD-stamp/XROOTD-install-Release.cmake:49 (message):
  Command failed: 1

   '/usr/local/bin/cmake' '-Dmake=' '-Dconfig=' '-P' '/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/builds/ROOT/build/builtins/xrootd/XROOTD-prefix/src/XROOTD-stamp/XROOTD-install-Release-impl.cmake'
```

### Suggested fix

https://github.com/SyneRBI/SIRF-SuperBuild/pull/898#issuecomment-2111226591

>I don't think we can have hinds falling regularly. But it's not great to have a lot of jobs falling silently either. So, I go back to my suggestion to install ROOT in another way for most jobs. Possibly, this is already done for the docker builds.

### PR DESCRIPTION
**Pull request by KrisThielemans**
Alternative to #898 

Fixes #899

### CODE REVIEW

**Comment by KrisThielemans** [comment]
@casperdcl the actions don't run. Presumably I have an error in the yml file, but I forgot how to find out. Can you help?

**Comment by KrisThielemans** [comment]
ok. well. I fixed the syntax, but now by condition doesn't work. I need help...

File: .github/workflows/c-cpp.yml [hunk 1]
```diff
@@ -81,6 +81,17 @@ jobs:
         sudo ./build_gadgetron-ubuntu.sh
         sudo ./build_python-ubuntu.sh
         echo "python-version=$(python3 --version | awk '{print $2}')" >> $GITHUB_OUTPUT
+
+    - if: contains(matrix.EXTRA_BUILD_FLAGS, fromJSON('["USE_SYSTEM_ROOT=ON"]'))
```
Comment (casperdcl) [inline .github/workflows/c-cpp.yml:85]:
```suggestion
    - if: contains(matrix.EXTRA_BUILD_FLAGS, "-DUSE_SYSTEM_ROOT=ON"))
```

File: .github/workflows/c-cpp.yml [hunk 2]
```diff
@@ -81,6 +81,17 @@ jobs:
...
+      name: install ROOT
+      run: |
+        # Install ROOT (warning: currently only valid on Ubuntu)
+        ROOT_file=root_v6.28.12.Linux-ubuntu20-x86_64-gcc9.4.tar.gz
+        wget https://root.cern/download/"$ROOT_file"
+        tar -xzvf "$ROOT_file"
+        rm "$ROOT_file"
+        source root/bin/thisroot.sh
```
Comment (casperdcl) [inline .github/workflows/c-cpp.yml:93]: Does this install to system binary dirs, or merely append to PATH?

**Review by casperdcl — CHANGES_REQUESTED** [review]

File: .github/workflows/c-cpp.yml (hunk 2 above)
Comment (KrisThielemans) [reply .github/workflows/c-cpp.yml:93]: That gives me
```
[Invalid workflow file: .github/workflows/c-cpp.yml#L85](https://github.com/SyneRBI/SIRF-SuperBuild/actions/runs/9125488514/workflow)
The workflow is not valid. .github/workflows/c-cpp.yml (Line: 85, Col: 11): Unexpected symbol: '"-DUSE_SYSTEM_ROOT'. Located at position 36 within expression: contains(matrix.EXTRA_BUILD_FLAGS, "-DUSE_SYSTEM_ROOT=ON"))
```

File: .github/workflows/c-cpp.yml (hunk 1 above)
Comment (KrisThielemans) [reply .github/workflows/c-cpp.yml]: [Invalid workflow file: .github/workflows/c-cpp.yml#L85](https://github.com/SyneRBI/SIRF-SuperBuild/actions/runs/9125488514/workflow)
The workflow is not valid. .github/workflows/c-cpp.yml (Line: 85, Col: 11): Unexpected symbol: '"-DUSE_SYSTEM_ROOT'. Located at position 36 within expression: contains(matrix.EXTRA_BUILD_FLAGS, "-DUSE_SYSTEM_ROOT=ON"))

File: .github/workflows/c-cpp.yml (hunk 2 above)
Comment (KrisThielemans) [reply .github/workflows/c-cpp.yml]: added them all to GITHUB_ENV

File: .github/workflows/c-cpp.yml (hunk 1 above)
Comment (KrisThielemans) [reply .github/workflows/c-cpp.yml]: I gave up on this and just did it in bash. Feel free to fix at some point!

**Pull request merged**